src/app/server/DefaultTermsAndConditionsProvider.cpp

```diff
@@ -0,0 +1,306 @@
+/*
+ *
+ *  Copyright (c) 2024 Project CHIP Authors
+ *  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "DefaultTermsAndConditionsProvider.h"
+#include "TermsAndConditionsProvider.h"
+
+#include <lib/core/CHIPError.h>
+#include <lib/core/TLV.h>
+#include <lib/core/TLVTypes.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/SafeInt.h>
+
+#define VerifyNoErrorOrReturnValue(expr, code, ...) VerifyOrReturnValue(CHIP_NO_ERROR == expr, code, ##__VA_ARGS__)
+#define VerifyNoErrorOrReturnInternal(expr, ...) VerifyNoErrorOrReturnValue(expr, CHIP_ERROR_INTERNAL, ##__VA_ARGS__)
+#define VerifyOrReturnInvalidArgument(expr, ...) VerifyOrReturnValue(expr, CHIP_ERROR_INVALID_ARGUMENT, ##__VA_ARGS__)
+#define VerifyOrReturnUninitialized(expr, ...) VerifyOrReturnValue(expr, CHIP_ERROR_UNINITIALIZED, ##__VA_ARGS__)
+#define VerifyOrReturnInternal(expr, ...) VerifyOrReturnValue(expr, CHIP_ERROR_INTERNAL, ##__VA_ARGS__)
+#define VerifyNotNullOrReturnInvalidArgument(expr, ...)                                                                            \
+    VerifyOrReturnValue(nullptr != expr, CHIP_ERROR_INVALID_ARGUMENT, ##__VA_ARGS__)
+#define VerifyNotNullOrReturnUninitialized(expr, ...) VerifyOrReturnValue(nullptr != expr, CHIP_ERROR_UNINITIALIZED, ##__VA_ARGS__)
+
+namespace {
+const chip::StorageKeyName kStorageKey                       = chip::DefaultStorageKeyAllocator::TermsAndConditionsAcceptance();
+constexpr chip::TLV::Tag kSerializationVersionTag            = chip::TLV::ContextTag(1);
+constexpr chip::TLV::Tag kAcceptedAcknowledgementsTag        = chip::TLV::ContextTag(2);
+constexpr chip::TLV::Tag kAcceptedAcknowledgementsVersionTag = chip::TLV::ContextTag(3);
+constexpr uint8_t kSerializationVersion                      = 1;
+
+constexpr size_t kEstimatedTlvBufferSize = chip::TLV::EstimateStructOverhead(sizeof(uint8_t),  // SerializationVersion
+                                                                             sizeof(uint16_t), // AcceptedAcknowledgements
+                                                                             sizeof(uint16_t)  // AcceptedAcknowledgementsVersion
+                                                                             ) *
```

Comment:
`* 8 * sizeof(uint16_t)` will lead to a 16x increase. I don't think you intended that...

Reply:
The issue is the multiplication by the 8 and the sizeof. I think you meant `+` at first. Bumping to 4 works.